### PR TITLE
docs: Examples for Route Matching

### LINKED ISSUE #5020: docs: need example to introduce the details of route match
### Issue description

Hi, I read the docs of https://apisix.apache.org/docs/apisix/architecture-design/router

And I think we can give some examples of this like:

1. Host + URI
2. Host + URI + Vars
3. Host + URI + Vars + FilterFunctions 

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

### PR DESCRIPTION
**Pull request by ro4i7**
### Description
added example to introduce the details of route match #5020; Issue description
examples of this like:

Host + URI
Host + URI + Vars
Host + URI + Vars + FilterFunctions;

Fixes #5020

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by starsz — COMMENTED** [review]
In fact, I want to know the priority of those route match type.

**Comment by ro4i7** [comment]
> In fact, I want to know the priority of those route match type.

Priority of those route match in order is : 

1. Host + URI + Vars + FilterFunctions: This criterion matches requests based on the combination of the host, URI, variables extracted from the URI using regular expressions, and the output of a user-defined filter function. This is the most specific and fine-grained type of route matching.

2. Host + URI + Vars: This criterion matches requests based on the combination of the host, URI, and variables extracted from the URI using regular expressions. This is less specific than the previous type but still allows for some level of variable-based matching.

3. Host + URI: This criterion matches requests based on the combination of the host and URI. This is the least specific type of route matching and will match any requests with the specified host and URI.

File: docs/en/latest/architecture-design/apisix.md [hunk 1]
```diff
@@ -49,3 +49,70 @@ The diagram below shows how APISIX handles an incoming request and applies corre
 The chart below shows the order in which different types of Plugin are applied to a request:
 
 ![flow-plugin-internal](https://raw.githubusercontent.com/apache/apisix/master/docs/assets/images/flow-plugin-internal.png)
+
+## Examples of route matching
+
+1. Host + URI: This criterion matches requests based on the combination of the host and URI. For example, you could use this to match all requests to https://example.com/api.
```
Comment (starsz) [inline docs/en/latest/architecture-design/apisix.md:55]: > match all requests to https://example.com/api ?

Or `https://example.com/foo/bar`?

File: docs/en/latest/architecture-design/apisix.md [hunk 2]
```diff
@@ -49,3 +49,70 @@ The diagram below shows how APISIX handles an incoming request and applies corre
...
+        "type": "roundrobin",
+        "nodes": {
+            "httpbin.org:80": 1
+        }
+    }
+}
+
+```
+
+2. Host + URI + Vars: This criterion matches requests based on the combination of the host, URI, and variables extracted from the URI using regular expressions. For example, you could use this to match all requests to https://example.com/api/users/123, where 123 is a variable extracted from the URI.
```
Comment (starsz) [inline docs/en/latest/architecture-design/apisix.md:71]:
```
https://example.com/api/users/123,
``` ? 
Are you sure ?

File: docs/en/latest/architecture-design/apisix.md [hunk 3]
```diff
@@ -49,3 +49,70 @@ The diagram below shows how APISIX handles an incoming request and applies corre
...
+        "type": "roundrobin",
+        "nodes": {
+            "httpbin.org:80": 1
+        }
+    }
+}
+
+```
+
+3. Host + URI + Vars + FilterFunctions: This criterion matches requests based on the combination of the host, URI, variables extracted from the URI using regular expressions, and the output of a user-defined filter function. For example, you could use this to match all requests to https://example.com/api/users/123 where the variable 123 represents a user ID that is valid in your application's database.
```
Comment (starsz) [inline docs/en/latest/architecture-design/apisix.md:90]: Ditto.

File: docs/en/latest/architecture-design/apisix.md (hunk 1 above)
Comment (ro4i7) [reply docs/en/latest/architecture-design/apisix.md:55]: I have written here by taking some random examples in all three cases. Please tell me about this.

File: docs/en/latest/architecture-design/apisix.md (hunk 3 above)
Comment (Sn0rt) [reply docs/en/latest/architecture-design/apisix.md:90]: @ro4i7 PTAL

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]
@ro4i7, your changes are inaccurate and IMO, these examples should be added here: https://github.com/apache/apisix/blob/master/docs/en/latest/terminology/route.md

**Pull request closed without merging**